projects/counters/src/main.c

```diff
@@ -0,0 +1,39 @@
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+static void prv_counter_callback(SoftTimerId soft_timer_id, void *context) {
+  Counters *counter_ptr = (Counters *)context;
+  uint8_t *counter_a_ptr = &(counter_ptr->counter_a);
```

Comment:
This seems sort of unneseccary.
Since we already have counter_ptr, which points to our  counter we created in main, we don't need a pointer to each section of the counter.
We just access/manipulate the data in there by using `counter_ptr->counter_a = ...`